crates/wac-resolver/src/lib.rs

```diff
@@ -56,22 +65,22 @@ pub enum Error {
         #[label(primary, "package `{name}` does not exist")]
         span: SourceSpan,
     },
-    /// The requested package version has been yanked.
+    /// The requested package version has been yanked or does not exist.
     #[cfg(feature = "registry")]
-    #[error("version {version} of package `{name}` has been yanked")]
-    PackageVersionYanked {
+    #[error("version {version} of package `{name}` has been yanked or does not exist")]
+    PackageVersionYankedOrDoesNotExist {
```

Comment:
Any reason to not have a separate variant here for `PackageVersionDoesNotExist` (or add an `Option<semver::Version>` to the `PackageDoesNotExist` variant)?

Reply:
At the moment, `download_exact()` returns an error that does distinguish between not found and yanked. That should be improved on the Warg side.

https://github.com/bytecodealliance/registry/blob/main/crates/client/src/lib.rs#L700-L713

Reply:
In that case, should we just call this `PackageVersionDoesNotExist` until we can distinguish?

Reply:
Done